library/java/org/chromium/net/impl/CronetUrlRequest.java

```diff
@@ -631,6 +653,19 @@ private boolean streamEnded() {
     return cronvoyCallbacks != null && cronvoyCallbacks.mEndStream;
   }
 
+  private void recordEnvoyFinalStreamIntel(EnvoyFinalStreamIntel envoyFinalStreamIntel) {
+    mEnvoyFinalStreamIntel = envoyFinalStreamIntel;
+    if (mUrlResponseInfo != null) { // Null if cancelled before receiving a Response.
+      mUrlResponseInfo.setReceivedByteCount(envoyFinalStreamIntel.getReceivedByteCount() +
```

Comment:
so I thought cronvoy had the following behavior
if we'd read 100 header bytes and 5000 body bytes from the network but reset the stream when only 2500 bytes had been sent up to the java layer then reset the stream, that we'd want to only report the 100+2500 bytes the java layer had seen?  I assumed you'd be doing the byte accounting of "headers + body bytes cronet had seen" in the java layers but it looks like we're reporting all bytes including buffered ones?  Is my understanding off?

Reply:
The `mEnvoyFinalStreamIntel` is expected to be null if the request is cancelled before receiving the ResponseHeader

Reply:
right, but if java gets the headers, and say half the body bytes which were read off the wire, what would cronet expect to report for bytes read: bytes read off the wire, or bytes handed up to java?

Reply:
still curious about this one

Reply:
That's a good question. There are no tests for that. Here is what I found after many hours of "cout": Cronet is not consistent when reporting the received byte count. There is one [test with an OK response](https://source.chromium.org/chromium/chromium/src/+/main:components/cronet/android/test/javatests/src/org/chromium/net/CronetUrlRequestTest.java;l=171) where all the bytes are counted, i.e. == `streamInfo().getUpstreamBytesMeter()->wireBytesReceived()` == 86, and there is another [test with a "NOT FOUND"](https://source.chromium.org/chromium/chromium/src/+/main:components/cronet/android/test/javatests/src/org/chromium/net/CronetUrlRequestTest.java;l=735) where the status line is excluded from received byte count, and the expected received byte count is 120 (should be 142)
 
I "cout"ed the Buffer in codec_impl.cc This is the content of the test with "an OK response":
```
=========
HTTP/1.1 200 OK^M
Connection: close^M
Content-Length: 3^M
Content-Type: text/plain^M
^M
GET
=========== size: 86
```
The size is perfectly matching the value in the original Cronet test.

This is the content of the test with a "NOT FOUND":
```
=========
HTTP/1.1 404 Not Found^M
Content-Length: 96^M
^M
<!DOCTYPE html>
<html>
<head>
<title>Not found</title>
<p>Test page loaded.</p>
</head>
</html>

=========== size: 142
```
But this one the original Cronet test expects 120 bytes: this looks like the size without the Status Line.

I'm not yet sure how to reconcile this.